src/ProxyClient/Fastly.php

```diff
@@ -0,0 +1,158 @@
+            'Accept' => 'application/json',
+        ];
+
+        if (true === $this->options['soft_purge']) {
+            $headers['Fastly-Soft-Purge'] = 1;
+        }
+
+        foreach (\array_chunk($tags, 256) as $tagChunk) {
+            $this->queueRequest(
+                Request::METHOD_POST,
+                sprintf('/service/%s/purge', $this->options['service_identifier']),
+                $headers + [
+                    // Can be changed to use json payload if queueRequest is changed to expose possibility to pass body
+                    'Surrogate-Key' => implode(' ', $tagChunk),
+                ],
+                false
+            );
+        }
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see https://docs.fastly.com/api/purge#soft_purge_0c4f56f3d68e9bed44fb8b638b78ea36
+     * @see https://docs.fastly.com/guides/purging/authenticating-api-purge-requests#purging-urls-with-an-api-token
+     */
+    public function purge($url, array $headers = [])
+    {
+        $headers['Fastly-Key'] = $this->options['authentication_token'];
+
+        if (true === $this->options['soft_purge']) {
+            $headers['Fastly-Soft-Purge'] = 1;
+        }
+
+        $this->queueRequest(
+            self::HTTP_METHOD_PURGE,
+            $url,
+            $headers,
+            false
+        );
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function refresh($url, array $headers = [])
+    {
+        $headers['Fastly-Key'] = $this->options['authentication_token'];
+
+        // First soft purge url
+        $this->queueRequest(
+            self::HTTP_METHOD_PURGE,
+            $url,
+            array_merge($headers, ['Fastly-Soft-Purge' => 1]),
+            false
+        );
+
+        // Secondly make sure refresh is triggered
+        $this->queueRequest(
+            Request::METHOD_HEAD,
+            $url,
+            $headers,
+            false
+        );
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see https://docs.fastly.com/api/purge#purge_bee5ed1a0cfd541e8b9f970a44718546
+     *
+     * Warning:
+     * - Does not support soft purge, for that use an "all" key.
+     * - Requires a API token of a user with at least Engineer permissions.
+     */
+    public function clear()
+    {
+        $headers = [
+            'Fastly-Key' => $this->options['authentication_token'],
+            'Accept' => 'application/json',
+        ];
+
+        $this->queueRequest(
+            Request::METHOD_POST,
+            sprintf('/service/%s/purge_all', $this->options['service_identifier']),
+            $headers,
+            false
+        );
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureOptions()
+    {
+        $resolver = parent::configureOptions();
+
+        $resolver->setRequired([
+            'authentication_token',
+            'service_identifier',
```

Comment:
i think we usually documented additional configuration options in phpdoc - could you check with the varnish client and document in a similar place?

Reply:
It’s in their docs as hard limit, I can add a link.

Reply:
for the varnish proxy client, we have this section in the phpdoc: https://github.com/FriendsOfSymfony/FOSHttpCache/blob/01d87faf5cc48e2164b958b0a64ed7ff2b8b71ea/src/ProxyClient/Varnish.php#L24, i'd like to do the same here. can be the same text as in the doc.